File: fixtures/projects/e2e-python/src/api/v1/handler.py
from src.types.api import APIRequest, APIResponse


def _normalize_path(path: str) -> str:
    return path.rstrip("/") or "/"


def handle_request(request: APIRequest) -> APIResponse:
    from src.auth.session import create_session

    path = _normalize_path(request.path)
    session = create_session(
        user_id="api-v1",
        email="v1@example.com",
        roles=["reader"],
    )
    payload = {"version": "v1", "path": path, "method": request.method}
    payload["token_prefix"] = session.token[:8]
    status = 200 if path.startswith("/api") else 404
    return APIResponse(status=status, body=payload, headers={"x-api-version": "v1"})

File: fixtures/projects/e2e-csharp/Assets/Scripts/API/V2/Handler.cs
using MyApp.Types;

namespace MyApp.API.V2
{
    public class Handler
    {
        public ApiResponse HandleRequest(string route)
        {
            if (route == "/status")
            {
                return new ApiResponse(true, "v2-online", 200);
            }

            return new ApiResponse(false, "v2-missing", 404);
        }

        public string VersionTag()
        {
            return "v2";
        }
    }
}
